ephios/plugins/baseshiftstructures/structure/named_teams.py

```diff
@@ -368,19 +368,47 @@ def _get_teams_with_users(self):
         teams_with_users += [(team, None) for team in team_by_uuid.values()]
         return teams_with_users
 
-    def get_participation_display(self):
-        confirmed_teams_with_users = self._get_teams_with_users()
-        participation_display = []
-        for team, users in confirmed_teams_with_users:
-            if users:
-                participation_display += [[user[0], user[1], team["title"]] for user in users]
-            if not users or len(users) < team["min_count"]:
-                required_qualifications = ", ".join(
-                    Qualification.objects.filter(pk__in=[team.get("qualification")]).values_list(
-                        "abbreviation", flat=True
-                    )
+    def get_list_export_data(self):
+        export_data = []
+        team_by_uuid = {team["uuid"]: team for team in self.configuration.teams}
+        for team_uuid, team in team_by_uuid.items():
+            missing_count = team["min_count"]
+            team_qualifications = (
+                Qualification.objects.filter(id=team["qualification"]) if team else []
+            )
+            for p in self.shift.participations.all():
+                participants_team_uuid = self._choose_team_for_participation(p)
+                if participants_team_uuid != team_uuid:
```

Comment:
wouldn't it be smarter to map the participations before the loop (similar to the team_by_uuid) to avoid the nested loop?